src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -152,37 +156,47 @@ def _to_df(self, agg_exprs: List[Expression]) -> DataFrame:
         aliased_agg = [_alias(a) for a in unique]
 
         if isinstance(self._group_type, _GroupByType):
-            return DataFrame(
-                self._df._session,
-                Aggregate(self._grouping_exprs, aliased_agg, self._df._plan),
+            group_plan = Aggregate(
+                self._grouping_exprs,
+                aliased_agg,
+                self._df._select_statement or self._df._plan,
             )
-        if isinstance(self._group_type, _RollupType):
-            return DataFrame(
-                self._df._session,
-                Aggregate(
-                    [Rollup(self._grouping_exprs)],
-                    aliased_agg,
-                    self._df._plan,
-                ),
+        elif isinstance(self._group_type, _RollupType):
+            group_plan = Aggregate(
+                [Rollup(self._grouping_exprs)],
+                aliased_agg,
+                self._df._select_statement or self._df._plan,
             )
-        if isinstance(self._group_type, _CubeType):
-            return DataFrame(
-                self._df._session,
-                Aggregate([Cube(self._grouping_exprs)], aliased_agg, self._df._plan),
+        elif isinstance(self._group_type, _CubeType):
+            group_plan = Aggregate(
+                [Cube(self._grouping_exprs)],
```

Comment:
I'm wondering if the following makes sense:

```
if pivot:
  group_plan = ...
else:
  if group by:
    grouping_exprs = self._grouping_exprs
  elif roll up:
    grouping_exprs = [Rollup(self._grouping_exprs)]
  elif cube:
    grouping_exprs = [Cube(self._grouping_exprs)]
  else:
    raise
  group_plan = Aggregate(grouping_exprs, aliased_agg,  self._df._select_statement or self._df._plan)

```

Reply:
hmm frankly I feel the current structure is more clear

Reply:
Yeah I'm mainly trying to see if we could reduce some code duplication, but cannot come up with anything better :( It's okay to merge as is.